lib/src/zenohUTransport.cpp

```diff
@@ -170,29 +167,27 @@ UStatus ZenohUTransport::send(const UUri &uri,
         return status;
     }
 
-    if (false == isRPCMethod(uri.resource())) {
-        status.set_code(sendPublish(uri, payload, attributes));
+    if (false == isRPCMethod(message.attributes().source().resource())) {
```

Comment:
you should check according to the UMessageType , and for the RPC also add a check if isRPCMethod